src/openms/include/OpenMS/FILTERING/CALIBRATION/InternalCalibration.h

```diff
@@ -70,402 +70,362 @@ namespace OpenMS
+      std::cout.precision(writtenDigits<double>(0.0));
+#endif
+      if (exp.empty())
+      {
+        std::cout << "Input is empty." << std::endl;
+        return;
+      }
 
-    /**
-     @brief Calibrate a peak map using given reference ids with one calibration function for the whole map.
+      if (exp[0].getType() != SpectrumSettings::PEAKS)
+      {
+        std::cout << "Attention: this function is assuming peak data." << std::endl;
+      }
 
-         Calibrate a map using given peptide identifications. The calibration function is calculated for the whole map.
-     The m/z-values of the reference identifications are calculated through the given sequence and charge of the peptide.
-     For the matching of the reference masses and the peaks the parameter mz_tolerance is used to
-     calculate a window around the reference masses. If more than one peak is found within this window the
-     closest peak is taken.
-     @param exp the uncalibrated peak map
-     @param calibrated_exp the calibrated peak map
-     @param ref_ids the reference peptide identifications
-     @param trafo_file_name file where the transformation function of the calibration is stored
-    */
-    template <typename InputPeakType>
-    void calibrateMapGlobally(const MSExperiment<InputPeakType> & exp, MSExperiment<InputPeakType> & calibrated_exp, std::vector<PeptideIdentification> & ref_ids, String trafo_file_name = "");
+      Size num_ref_peaks = ref_masses.size();
+      bool use_ppm = (param_.getValue("mz_tolerance_unit") == "ppm") ? true : false;
+      double mz_tol = param_.getValue("mz_tolerance");
+      startProgress(0, exp.size(), "calibrate spectra");
+      // for each spectrum
+      for (Size spec = 0; spec < exp.size(); ++spec)
+      {
+        // calibrate only MS1 spectra
+        if (exp[spec].getMSLevel() != 1)
+        {
+          continue;
+        }
 
-    /**
-     @brief Calibrate an annotated feature map with one calibration function for the whole map.
 
-         Calibrate an annotated (!) feature map using the features' identifications. The calibration function is calculated for the whole map. The m/z-values of the reference identifications are calculated through the given sequence and charge of the peptide.
-     @param feature_map the uncalibrated feature map (annotated with peptide ids)
-     @param calibrated_feature_map the calibrated feature map
-     @param trafo_file_name file where the transformation function of the calibration is stored
-    */
-    void calibrateMapGlobally(const FeatureMap & feature_map, FeatureMap & calibrated_feature_map, String trafo_file_name = "");
+        std::vector<double> corr_masses, found_ref_masses;
+        UInt corr_peaks = 0;
+        for (Size peak = 0; peak <  exp[spec].size(); ++peak)
+        {
+          for (Size ref_peak = 0; ref_peak < num_ref_peaks; ++ref_peak)
+          {
+            if (!use_ppm && fabs(exp[spec][peak].getMZ() - ref_masses[ref_peak]) <  mz_tol)
```

Comment:
there seems to be some code duplication. One could calculate a const double absolute_mz_tolerance = use_ppm ? ... : mz_tol; outside of the loop (for both ppm and absolute tolerance).